internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -829,50 +880,90 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	if d.HasChange("paused") && !d.Get("paused").(bool) {
 		cluster.Paused = conversion.Pointer(d.Get("paused").(bool))
 	}
+	return cluster, nil
+}
 
-	timeout := d.Timeout(schema.TimeoutUpdate)
+func updateRequestOldAPI(d *schema.ResourceData, clusterName string) (*admin20231115.AdvancedClusterDescription, diag.Diagnostics) {
+	cluster := new(admin20231115.AdvancedClusterDescription)
 
-	if d.HasChange("advanced_configuration") {
-		ac := d.Get("advanced_configuration")
-		if aclist, ok := ac.([]any); ok && len(aclist) > 0 {
-			params := expandProcessArgs(d, aclist[0].(map[string]any))
-			if !reflect.DeepEqual(params, admin20231115.ClusterDescriptionProcessArgs{}) {
-				_, _, err := connV220231115.ClustersApi.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, &params).Execute()
-				if err != nil {
-					return diag.FromErr(fmt.Errorf(errorConfigUpdate, clusterName, err))
-				}
-			}
-		}
+	if d.HasChange("backup_enabled") {
+		cluster.BackupEnabled = conversion.Pointer(d.Get("backup_enabled").(bool))
 	}
 
-	// Has changes
-	if !reflect.DeepEqual(cluster, clusterChangeDetect) {
-		err := retry.RetryContext(ctx, timeout, func() *retry.RetryError {
-			_, resp, err := updateAdvancedCluster(ctx, connV220231115, connV2, cluster, projectID, clusterName, timeout)
-			if err != nil {
-				if resp == nil || resp.StatusCode == 400 {
-					return retry.NonRetryableError(fmt.Errorf(errorUpdate, clusterName, err))
-				}
```

Comment:
Removed this retry logic that was being done on top of calling the update. By looking into `updateAdvancedCluster` it clear that when err != nil, resp is always nil, meaning it always went into the NonRetryableError.